handlers/auth_handler_test.go

```
@@ -0,0 +1,862 @@
+	var bodyReader *bytes.Reader
+	if body != nil {
+		jsonBody, err := json.Marshal(body)
+		if err != nil {
+			t.Fatalf("failed to marshal request body: %v", err)
+		}
+		bodyReader = bytes.NewReader(jsonBody)
+	} else {
+		bodyReader = bytes.NewReader([]byte{})
+	}
+
+	req, err := http.NewRequest(method, url, bodyReader)
```

golangci-lint (2.11.4)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)